Servers/UI/OJS.Servers.Ui/ClientApp/src/pages/contests/ContestsPage.tsx

```diff
@@ -164,31 +172,68 @@ const ContestsPage = () => {
         [ contests, currentPage, handlePageChange, isLoaded, pagesInfo, renderContest ],
     );
 
+    const updateBreadcrumbAndNavigateToCategory = useCallback(
+        (breadcrumb: ICategoriesBreadcrumbItem) => {
+            const category = categoriesFlat.find(({ id }) => id.toString() === breadcrumb.id.toString());
+
+            updateBreadcrumb(category, categoriesFlat);
+            navigate(getContestCategoryBreadcrumbItemPath(breadcrumb.id));
+        },
+        [ categoriesFlat, navigate, updateBreadcrumb ],
+    );
+
+    const renderCategoriesBreadcrumbItem = useCallback(
+        (categoryBreadcrumbItem: ICategoriesBreadcrumbItem) => {
+            const { value, isLast } = categoryBreadcrumbItem;
+            const classNames = concatClassNames(styles.breadcrumbBtn, isLast
+                ? styles.breadcrumbBtnLast
+                : '');
+
+            return (
+                <Button
+                  type={ButtonType.plain}
+                  className={classNames}
+                  onClick={() => updateBreadcrumbAndNavigateToCategory(categoryBreadcrumbItem)}
+                  text={value}
+                />
+            );
+        },
+        [ updateBreadcrumbAndNavigateToCategory ],
+    );
+
     useEffect(
         () => {
             if (!areQueryParamsValid() && isLoaded) {
                 setShowAlert(true);
                 clearParams();
             }
         },
-        [ areQueryParamsValid, isLoaded, clearParams ],
+        [
+            areQueryParamsValid,
+            isLoaded,
+            clearParams,
+            breadcrumbItems,
+            handleFilterClick,
+            renderCategoriesBreadcrumbItem,
+            renderContests,
+        ],
     );
 
     return (
         <>
             {contestsAreLoading && <div style={{ ...flexCenterObjectStyles }}><SpinningLoader /></div>}
             {showAlert &&
-              (
-                  <Alert
-                    message="The category you requested was not valid, all contests were loaded."
-                    severity={AlertSeverity.Error}
-                    variant={AlertVariant.Filled}
-                    autoHideDuration={3000}
-                    vertical={AlertVerticalOrientation.Bottom}
-                    horizontal={AlertHorizontalOrientation.Right}
-                  />
-              )}
-            <ContestBreadcrumb />
+                (
+                    <Alert
+                      message="The category you requested was not valid, all contests were loaded."
+                      severity={AlertSeverity.Error}
+                      variant={AlertVariant.Filled}
+                      autoHideDuration={3000}
+                      vertical={AlertVerticalOrientation.Bottom}
+                      horizontal={AlertHorizontalOrientation.Right}
+                    />
+                )}
+            <Breadcrumb items={breadcrumbItems} itemFunc={renderCategoriesBreadcrumbItem} />
```

Comment:
Why the spacing is different here ?

Reply:
Not sure what caused this, but it got automatically fixed after saving the file.